tests/test-customize-changesets-controller.php

```diff
@@ -99,29 +518,1239 @@ public function test_update_item() {
+		}
+
+		wp_set_current_user( self::$admin_id );
+
+		$uuid = wp_generate_uuid4();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertSame( 'publish', $data['status'] );
+	}
+
+	/**
+	 * Test that publishing a changeset with update_item() returns a 'publish' status.
+	 */
+	public function test_update_item_edit_status_is_publish_after_publish() {
+		if ( post_type_supports( 'customize_changeset', 'revisions' ) ) {
+			$this->markTestSkipped( 'Changesets are not trashed when revisions are enabled.' );
+		}
+
+		wp_set_current_user( self::$admin_id );
+
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$data = $response->get_data();
+		$this->assertSame( 'publish', $data['status'] );
+	}
+
+	/**
+	 * Test that creating a published changeset with update_item() returns a new changeset UUID.
+	 */
+	public function test_update_item_create_has_next_changeset_id_after_publish() {
+		wp_set_current_user( self::$admin_id );
+
+		$uuid_before = wp_generate_uuid4();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid_before ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$data = $response->get_data();
+		$this->assertTrue( isset( $data['next_changeset_uuid'] ) );
+		$this->assertNotSame( $data['next_changeset_uuid'], $uuid_before );
+	}
+
+	/**
+	 * Test that publishing an existing changeset with update_item() returns a new changeset UUID.
+	 */
+	public function test_update_item_edit_has_next_changeset_id_after_publish() {
+		wp_set_current_user( self::$admin_id );
+
+		$uuid_before = wp_generate_uuid4();
+
+		$manager = new WP_Customize_Manager( array(
+			'changeset_uuid' => $uuid_before,
+		) );
+		$manager->save_changeset_post();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid_before ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$data = $response->get_data();
+		$this->assertTrue( isset( $data['next_changeset_uuid'] ) );
+		$this->assertNotSame( $data['next_changeset_uuid'], $uuid_before );
+	}
+
+	/**
+	 * Test that creating a published changeset with update_item() updates a valid setting.
+	 */
+	public function test_update_item_create_with_bloginfo() {
+		wp_set_current_user( self::$admin_id );
+
+		$uuid = wp_generate_uuid4();
+		$blogname_after = rand_str();
```

Comment:
I recall random strings being frowned upon now in unit tests, though I realize that is what the UUID4 will be. It would probably be better to use a literal string for the blogname.